### PR TITLE
Unit testcases for user and transactions has been implemented in User

### PR DESCRIPTION
**Pull request by jamitra**

### CODE REVIEW

File: src/shared/exception-handler.ts [hunk 1]
```diff
@@ -6,50 +6,34 @@ import {
   Catch,
   ArgumentsHost,
   HttpStatus,
+  BadRequestException,
 } from '@nestjs/common';
 import { HttpAdapterHost } from '@nestjs/core';
 
-@Catch()
+@Catch(BadRequestException)
```
Comment (shashanktiwari-hashedin) [inline src/shared/exception-handler.ts:13]: we can't add BadRequestException here if we add this then only in case of 500 this will work not for 400 or 401 etc.

File: src/shared/exception-handler.ts [hunk 2]
```diff
@@ -6,50 +6,34 @@ import {
...
 export class AllExceptionsFilter implements ExceptionFilter {
   constructor(private readonly httpAdapterHost: HttpAdapterHost) { }
 
-  catch(exception: any, host: ArgumentsHost): void {
+  catch(exception: BadRequestException, host: ArgumentsHost) {
```
Comment (shashanktiwari-hashedin) [inline src/shared/exception-handler.ts:17]: this exception can't be of type BadRequest it must be of type any as declaring it of type Bad request it will not take other exceptions.

File: src/shared/exception-handler.ts [hunk 3]
```diff
@@ -6,50 +6,34 @@ import {
...
     const ctx = host.switchToHttp();
     const response = ctx.getResponse();
 
+    let status = exception instanceof BadRequestException
```
Comment (shashanktiwari-hashedin) [inline src/shared/exception-handler.ts:21]: this change is not correct.

File: src/shared/network.ts [hunk 4]
```diff
@@ -9,8 +9,8 @@
  import { API_TIMEOUT_VALUE } from '../commons/constant';
  import { isEmpty } from '../commons/util';
  const BASE_URL = ''; // BASE_URL is the base endpoint at which API is hit.
- 
- export function postRequest(postData: any, headerData: any, endPoint: string, skipBaseUrl: boolean) {
+ export default class Network {
```
Comment (shashanktiwari-hashedin) [inline src/shared/network.ts:12]: this shouldn't be made as a class we can add normal unit test for functions keep them as functions only.

File: package.json [hunk 5]
```diff
@@ -15,6 +15,10 @@
         "start:prod": "node dist/main",
         "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
         "test": "jest",
+        "test:product": "jest ./src/product",
```
Comment (shashanktiwari-hashedin) [inline package.json:18]: i think these commands are not needed instead this sample should be put in read me file if already present in read me that is sufficient , remove from here.

File: package.json [hunk 6]
```diff
@@ -26,7 +30,10 @@
         "@nestjs/platform-express": "^8.0.0",
         "@nestjs/platform-fastify": "^8.2.3",
         "@nestjs/sequelize": "^8.0.0",
+        "@types/chai-as-promised": "^7.1.4",
         "axios": "^0.24.0",
+        "chai": "^4.3.4",
```
Comment (shashanktiwari-hashedin) [inline package.json:35]: are we using chai?

Comment (shashanktiwari-hashedin) [reply package.json:35]: mention the file name where we have used chai.

File: src/shared/exception-handler.spec.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        jest.clearAllMocks();
+        const module: TestingModule = await Test.createTestingModule({
+            providers: [
+                AllExceptionsFilter,
+            ]
+        }).compile();
+        service = module.get<AllExceptionsFilter>(AllExceptionsFilter);
+    });
+
+    describe('All exception filter tests', () => {
```
Comment (shashanktiwari-hashedin) [inline src/shared/exception-handler.spec.ts:53]: just make it exception filter tests.

File: package.json (hunk 6 above)
Comment (jamitra) [reply package.json:35]: Chai is used in Validator-pipe test.

File: src/shared/logger.services.spec.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+      json: jest.fn()
+    },
+    createLogger: jest.fn().mockReturnValue(logger),
+    transports: {
+      Console: jest.fn(),
+      File: jest.fn()
+    }
+  }));
+  
+import { CartsService } from "src/carts/services/carts.service";
```
Comment (shashanktiwari-hashedin) [inline src/shared/logger.services.spec.ts:22]: add import at the top .

File: src/shared/logger.services.spec.ts [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  import * as winston from "winston";
+  import { LoggerService } from "./logger.service";
+  
+  describe("-- Logging Service --", () => {
+    let loggerMock: winston.Logger;
+  
+    test("testing logger log function called...", () => {
+      const mockCreateLogger = jest.spyOn(winston, "createLogger");
+      loggerMock = mockCreateLogger.mock.instances[0];
+    //   expect(loggingService).toBeInstanceOf(LoggerService);
```
Comment (shashanktiwari-hashedin) [inline src/shared/logger.services.spec.ts:32]: do we need commented code?

File: src/shared/logger.services.spec.ts [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+      expect(logger.debug).toHaveBeenCalled();
+      logger.warn("testing warn");
+      expect(logger.warn).toHaveBeenCalled();
+      logger.verbose("testing verbose");
+      expect(logger.verbose).toHaveBeenCalled();
+      logger.error("testing the error");
+      expect(logger.error).toHaveBeenCalled();
+  
+      // now try and invoke the logger instance indirectly through the service class
+      // check that loggerMock is called a second time - this fails, only called once
```
Comment (shashanktiwari-hashedin) [inline src/shared/logger.services.spec.ts:50]: ?

File: src/shared/logger.services.spec.ts [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+      // from the preceding lines in this test
+    //   LoggerService.debug("debug message");
+  
+    //   expect(logger.debug).toHaveBeenCalledTimes(1); // <- here
+    });
+  });
+
```
Comment (shashanktiwari-hashedin) [inline src/shared/logger.services.spec.ts:57]: line at the end of file is missing.

File: src/shared/network.spec.ts [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+         headers:{},
+         timeout: API_TIMEOUT_VALUE,
+        }));
+        expect(putRequestSpy).toHaveBeenCalledTimes(1);
+    }catch(err) {
+        throw Error("Error in putRequest testcase");
+    }
+    })
+})
+
```
Comment (shashanktiwari-hashedin) [inline src/shared/network.spec.ts:56]: end of file line is missing.

File: src/shared/validation.pipe.spec.ts [hunk 13]
```diff
@@ -0,0 +1,174 @@
+import { ArgumentMetadata } from '@nestjs/common';
+import * as chai from 'chai';
+import { expect } from 'chai';
```
Comment (shashanktiwari-hashedin) [inline src/shared/validation.pipe.spec.ts:3]: jest provides the same functionality why do we need to use chai for that?

File: src/user/controllers/user.controller.spec.ts [hunk 14]
```diff
@@ -67,7 +67,7 @@ describe('UserController', () => {
   describe('create()', () => {
     it('should create a user', async () => {
       expect(
-        await userController.createUser({
+        await userController.addUser({
```
Comment (shashanktiwari-hashedin) [inline src/user/controllers/user.controller.spec.ts:70]: revert these changes these have been done in another pr it will cause merge conflicts.
